app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java

```diff
@@ -77,9 +77,9 @@ public int getItemCount() {
     }
 
     public void removeAt(@NonNull final int position) {
-        albumList.remove(position);
+        albums.remove(position);
         notifyItemRemoved(position);
-        notifyItemRangeChanged(0, albumList.size());
+        notifyItemRangeChanged(0, albums.size());
```

Comment:
`notifyItemRangeChanged(0, albums.size())` is the same as `notifyDataSetChanged` and it tells the adapter that the data set has changed and all the views are refreshed. However `notifyItemRangeChanged` can be used to update only the first 2 items for example and this way you avoid refreshing all the views.

Same deal with `notifyItemRemoved`. This is just a way to improve performance so not all views need to be refreshed.

So doing `notifyItemRemoved` should be enough in this case